hudhook/src/hooks/common.rs

```diff
@@ -0,0 +1,119 @@
+use imgui::*;
+use parking_lot::MutexGuard;
+use windows::Win32::Foundation::{HWND, LPARAM, LRESULT, WPARAM};
+use windows::Win32::UI::Input::KeyboardAndMouse::*;
+use windows::Win32::UI::WindowsAndMessaging::{WHEEL_DELTA, WM_XBUTTONDBLCLK, XBUTTON1, *};
+
+pub(crate) type WndProcType =
+    unsafe extern "system" fn(hwnd: HWND, umsg: u32, wparam: WPARAM, lparam: LPARAM) -> LRESULT;
+
+pub(crate) trait ImguiRendererCommon {
```

Comment:
I'd change the `Common` suffix, or the name of the trait altogether, as I feel the current one doesn't communicate its intentions well enough. This is a nitpick, though, I don't have good alternative ideas for the time being so we can postpone it, or even merge and change it in another PR.

Reply:
hmm, any suggestion? I'd guess `ImguiRendererInterface` could be more appropriated? still unsure how to name it hehe.

Reply:
I read a bit from [here](https://github.com/rust-lang/api-guidelines/discussions/28) and then I realized we may go with a more radical approach here. The role of this trait is probably not limited to factoring out common behavior for DX impls but more generally could be the entry point to `WndProc` and other events related considerations.

I'd go for something like `WindowsEventHandler` (though I don't super like this either so feel free to come up with other suggestions -- it's an internal so we're never too late to change it). This would also give us a good spot where to put event related behavior extensions in the future (like `focus`).